andela/jubilee-backend
# #168132789 Build out the Signup as a company endpoint

#### What does this PR do?
Enable companies to be able to signup to use the platform

#### Description of Task to be completed?
- Create the database models, migrations and establish appropriate relationships between tables
- Create an endpoint for registering suppliers
- Validate all user inputs before registration
- Implement middleware for checking if the user already exists
- Ensure that an email gets fired upon successful signup
- Write unit tests for this feature
- Document the new endpoint on swagger
#### How should this be manually tested?
- Clone the repo and cd into the apps working directory
- Checkout to the **ft-company-signup-168132789** branch
- Create a .env file and specify the variables described in the .env.sample inside this file.
- Run docker-compose up --build to build the docker image and spin up all the necessary services
- Alternative to running the app on docker, you can run `npm run start:dev`
- Run `npm run migrate:reset` && `npm run migrate` && `npm run seed` to create tables in your 
 database and seed the database tables with the necessary app configs.
- Launch Postman/Insomnia and make a POST request to localhost:3000/api/auth/signup/company 
 route using the fields described in the swagger document of this project.
- Upon successful registration, a JSON response should be received with updated user parameters.
- To run the unit tests, run `npm run test`.
#### Any background context you want to provide?
This application is bootstrapped using docker as described above, however, the app can still be run locally without docker using the npm run start:dev script

#### What are the relevant pivotal tracker stories?
[#168132789](https://www.pivotaltracker.com/story/show/168132789)
